src/main/java/com/exactpro/th2/FixHandler.java

```diff
@@ -606,6 +698,26 @@ public void sendLogon() {
         channel.send(Unpooled.wrappedBuffer(logon.toString().getBytes(StandardCharsets.UTF_8)), Collections.emptyMap(), null, IChannel.SendMode.MANGLE);
     }
 
+    private void sendLogout() {
+        StringBuilder logout = new StringBuilder();
+        setHeader(logout, MSG_TYPE_LOGOUT, msgSeqNum.incrementAndGet(), false);
+        setChecksumAndBodyLength(logout);
+
+        if (enabled.get()) {
+            channel.send(Unpooled.wrappedBuffer(logout.toString().getBytes(StandardCharsets.UTF_8)), Collections.emptyMap(), null, IChannel.SendMode.MANGLE);
+        }
+        disconnectRequest = executorService.schedule(() -> {
+            enabled.set(false);
+            try {
+                Util.writeSequences(msgSeqNum.get(), serverMsgSeqNum.incrementAndGet(), settings.getStateFilePath());
+            } catch (IOException e) {
+                if (LOGGER.isErrorEnabled()) {
+                    LOGGER.error("Sequence number were not saved: clientSeq - {}, serverSeq - {}", msgSeqNum.get(), serverMsgSeqNum.get());
+                }
+            }
```

Comment:
Why are we delaying it? This delay could prevent us from saving sequences in time when this method is called from `close` method